lib/orderbook/OrderBook.ts

```diff
@@ -201,9 +204,20 @@ class OrderBook extends EventEmitter {
     const pair = this.pairs.get(pairId);
     if (pair) {
       this.pairs.delete(pairId);
-      this.matchingEngines.delete(pairId);
-      // TODO: invalidate all orders for this pair
-      // TODO: update handshake state
+      const matchingEngine = this.matchingEngines.get(pairId);
+
+      if (matchingEngine) {
```

Comment:
should we assert this, instead of having a silent error?

we also have `getMatchingEngine` which throws exception, but in this case this error is not expected to ever happen.